packages/functional-tests/tests/misc/changePasswordKeyStretchingV2.spec.ts

```diff
@@ -0,0 +1,135 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { expect, test } from '../../lib/fixtures/standard';
+import { BaseTarget } from '../../lib/targets/base';
+
+const AGE_21 = '21';
+
+/**
+ * These tests represent various permutations between interacting with V1 and V2
+ * key stretched passwords. We need to ensure that operations are interchangeable!
+ */
+test.describe('severity-2 #smoke', () => {
+  test.describe('key-stretching-v2', () => {
+    // Helpers
+    async function _getKeys(
+      version: 1 | 2,
+      target: BaseTarget,
+      email: string,
+      password: string
+    ) {
+      const client = target.createAuthClient(version);
+      const response = await client.signIn(email, password, { keys: true });
+      const keys = client.accountKeys(
+        response.keyFetchToken,
+        response.unwrapBKey
+      );
+      return keys;
+    }
+
+    type TestCase = {
+      signup: { version: 1 | 2; query: string };
+      change: { version: 1 | 2; query: string };
+    };
+    const TestCases: TestCase[] = [
```

Comment:
Perhaps these should be programmatically generated. It might reduce the risk of typo, and ensure all cases get hit. (Not that I noticed any typos or missing cases.)

It also seems like the state of query is one to with version, so maybe there's not a reason to specify it?

Reply:
```js
const v1 = { version: 1, query: '' };
const v2 = { version: 2, query: 'stretch=2' };

const tuples = [
  [v1, v1],
  [v1, v2],
  [v2, v1],
  [v2, v2],
];

const TestCases = [];
for (const [t1, t2] of tuples) {
  TestCases.push({ signup: t1, change: t2 });
}

console.log(TestCases);
```

OUTPUT
```js
[
  {
    signup: { version: 1, query: '' },
    change: { version: 1, query: '' }
  },
  {
    signup: { version: 1, query: '' },
    change: { version: 2, query: 'stretch=2' }
  },
  {
    signup: { version: 2, query: 'stretch=2' },
    change: { version: 1, query: '' }
  },
  {
    signup: { version: 2, query: 'stretch=2' },
    change: { version: 2, query: 'stretch=2' }
  }
]
```

Reply:
This is much better! Made the change --> https://github.com/mozilla/fxa/pull/16923/commits/420b916a16c52e83713efba7b59484860d86164d